Repaint rects for slider controls are incorrect for RTL vertical block writing modes

lilyspiniolas opened the pull request:
#### 8896aad97fa3f274318189a550b3e0abb25780ba
<pre>
Repaint rects for slider controls are incorrect for RTL vertical block writing modes
<a href="https://bugs.webkit.org/show_bug.cgi?id=297821">https://bugs.webkit.org/show_bug.cgi?id=297821</a>
<a href="https://rdar.apple.com/158567821">rdar://158567821</a>

Reviewed by Aditya Keerthi.

For slider elements with RTL vertical block writing modes, repaint the input
before layout to avoid an issue where the repaint rect is positioned incorrectly.
The issue is suspected to stem from the following bug <a href="https://webkit.org/b/70762.">https://webkit.org/b/70762.</a>

Move `RenderThemeMac::adjustRepaintRect` to `RenderThemeCocoa::adjustRepaintRect`
so that the repaint rect is properly inflated to include the thumb&apos;s shadow.

* LayoutTests/fast/forms/range/slider-repaint-vertical-rl-expected.txt: Added.
* LayoutTests/fast/forms/range/slider-repaint-vertical-rl.html: Added.
* LayoutTests/platform/ios-18/fast/forms/range/slider-repaint-vertical-rl-expected.txt: Added.
* LayoutTests/platform/ios/TestExpectations:
* LayoutTests/platform/ios/fast/forms/range/slider-repaint-vertical-rl-expected.txt: Added.
* LayoutTests/platform/mac-sequoia/fast/forms/range/slider-repaint-vertical-rl-expected.txt: Added.
* LayoutTests/platform/mac-sonoma/fast/forms/range/slider-repaint-vertical-rl-expected.txt: Added.
* LayoutTests/platform/mac/fast/forms/range/slider-repaint-vertical-rl-expected.txt: Added.
* Source/WebCore/html/shadow/SliderThumbElement.cpp:
(WebCore::SliderThumbElement::setPositionFromPoint):
* Source/WebCore/rendering/cocoa/RenderThemeCocoa.h:
* Source/WebCore/rendering/cocoa/RenderThemeCocoa.mm:
(WebCore::RenderThemeCocoa::adjustRepaintRect):
* Source/WebCore/rendering/mac/RenderThemeMac.h:
* Source/WebCore/rendering/mac/RenderThemeMac.mm:
(WebCore::RenderThemeMac::adjustRepaintRect): Deleted.

Canonical link: <a href="https://commits.webkit.org/299690@main">https://commits.webkit.org/299690@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/a0565a55ee09bc9020e35fffb4f18b6d3138a912

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/119886 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/39579 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/30230 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/126202 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/71959 "Built successfully") | ⏳ 🛠 ios-apple 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/121762 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/40274 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/48156 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/91039 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/60346 "Passed tests") | ⏳ 🛠 mac-apple 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/122838 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/32157 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/107493 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/71595 "Passed tests") | | ⏳ 🛠 vision-apple 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/31187 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/25599 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/69847 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/101629 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/25789 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/129132 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/46806 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/35476 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/99660 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/47172 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/103684 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/99505 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/25257 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/44948 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/22961 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/43412 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/46668 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/52374 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/46134 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/49483 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/47820 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

pxlcoder reviewed (commented):
Can we add tests for this? Maybe similar to the ones I added in https://commits.webkit.org/272799@main.

lilyspiniolas committed:
Repaint rects for slider controls are incorrect for RTL vertical block writing modes
https://bugs.webkit.org/show_bug.cgi?id=297821
rdar://158567821

Reviewed by Aditya Keerthi.

For slider elements with RTL vertical block writing modes, repaint the input
before layout to avoid an issue where the repaint rect is positioned incorrectly.
The issue is suspected to stem from the following bug https://webkit.org/b/70762.

Move `RenderThemeMac::adjustRepaintRect` to `RenderThemeCocoa::adjustRepaintRect`
so that the repaint rect is properly inflated to include the thumb's shadow.

* LayoutTests/fast/forms/range/slider-repaint-vertical-rl-expected.txt: Added.
* LayoutTests/fast/forms/range/slider-repaint-vertical-rl.html: Added.
* LayoutTests/platform/ios-18/fast/forms/range/slider-repaint-vertical-rl-expected.txt: Added.
* LayoutTests/platform/ios/TestExpectations:
* LayoutTests/platform/ios/fast/forms/range/slider-repaint-vertical-rl-expected.txt: Added.
* LayoutTests/platform/mac-sequoia/fast/forms/range/slider-repaint-vertical-rl-expected.txt: Added.
* LayoutTests/platform/mac-sonoma/fast/forms/range/slider-repaint-vertical-rl-expected.txt: Added.
* LayoutTests/platform/mac/fast/forms/range/slider-repaint-vertical-rl-expected.txt: Added.
* Source/WebCore/html/shadow/SliderThumbElement.cpp:
(WebCore::SliderThumbElement::setPositionFromPoint):
* Source/WebCore/rendering/cocoa/RenderThemeCocoa.h:
* Source/WebCore/rendering/cocoa/RenderThemeCocoa.mm:
(WebCore::RenderThemeCocoa::adjustRepaintRect):
* Source/WebCore/rendering/mac/RenderThemeMac.h:
* Source/WebCore/rendering/mac/RenderThemeMac.mm:
(WebCore::RenderThemeMac::adjustRepaintRect): Deleted.

Canonical link: https://commits.webkit.org/299690@main
